### PR TITLE
Test recompilation

### PR DESCRIPTION
**Pull request by GuillaumeGen**
This adds a test for #1771 as suggested by #1783

### CODE REVIEW

File: tests/recompilation/BUILD.bazel [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+rules_haskell_integration_test(
+    name = "recompilation_test",
+    srcs = ["SomeTest.hs"],
+    workspace_path = "recompilation_workspace",
+    deps = [
+        "//tests/hackage:process",
+    ]
+)
+
+package(default_testonly = 1)
```
Comment (GuillaumeGen) [inline tests/recompilation/BUILD.bazel:12]: This line is is required due to to an attribute missing in the integration test framework provided by Bazel contribs, as explained in this pull request https://github.com/bazel-contrib/rules_bazel_integration_test/pull/72
Thank to @k1nkreet for spotting it.

Comment (cgrindel) [reply tests/recompilation/BUILD.bazel:12]: FYI. [Release 0.18.0](https://github.com/bazel-contrib/rules_bazel_integration_test/releases/tag/v0.18.0) of `rules_bazel_integration_test` contains the fix from https://github.com/bazel-contrib/rules_bazel_integration_test/pull/72 (via https://github.com/bazel-contrib/rules_bazel_integration_test/pull/212). You should not need the `package(default_testonly = 1)`. If you run into any issues, please let me know.

Comment (aherrmann) [reply tests/recompilation/BUILD.bazel]: Thank you! I've rebased, updated, and this seems to be working!

**Review by aherrmann — APPROVED** [review]
LGTM!

**Pull request merged**